data_warehouse/services/aggregate_data_service.py

```diff
@@ -355,8 +362,36 @@ def table_coverage_statistics(cls, context):
         return file_path, '专任教师培训覆盖率.xls'
 
     @classmethod
+    @admin_required()
     def traning_feedback(cls, context):
         '''培训记录反馈导出'''
+        request = context.get('request')
+        program_id = context.get('program_id', None)
+        deparment_id = context.get('department_id', None)
+        if not (program_id or deparment_id):
+            raise BadRequest('必须给定部门ID或者项目ID')
+        elif not deparment_id:
+            program_ids = [program_id]
```

Comment:
Here, your program_id is a string, consider casting it into a integer

Reply:
I will leave this job to upper caller.